### PR TITLE
[APPS-3537] Product specific logos and icons

### PR DESCRIPTION
**Pull request by mtsolakiszen**
🐻 

/cc @zendesk/vegemite @zendesk/wombat 

### Description

This PR makes ZAS handle product specific logos and icons. A `logoAssetHash` has been added to the app which has key: product name and value: logo path.

For multiproduct apps the logo will be under `assets/{product_name}/logo-small.png` and for single product apps it will be under `assets/logo-small.png`.

The same applies to the location icon assets which were already being passed as a hash on the app. 

### References
* JIRA: https://zendesk.atlassian.net/browse/APPS-3537
* ZAM PR: https://github.com/zendesk/zendesk_app_market/pull/2760
* ZAF PR: https://github.com/zendesk/zendesk_app_framework/pull/759
* Zendesk console PR: https://github.com/zendesk/zendesk_console/pull/14109

### Risks
* [medium] ZAS incorrectly parses app assets during the build process

### CODE REVIEW

File: lib/zendesk_apps_support/package.rb [hunk 1]
```diff
@@ -289,34 +299,35 @@ def location_icons
...
           location = location_options.location.name
-          location_icons[host][location] = build_location_icons_hash(location)
+          product_prefix = manifest.products.count > 1 ? "#{host}/" : ''
+          location_icons[host][location] = build_location_icons_hash(location, product_prefix)
         end
       end
     end
 
-    def build_location_icons_hash(location)
+    def build_location_icons_hash(location, product_prefix)
```
Comment (seangoedecke) [inline lib/zendesk_apps_support/package.rb]: nit: I'm not sure `product_prefix` makes sense as an argument name here - it's in the middle of the icon url, after all, not at the start. Maybe `product_dir`?

Comment (mtsolakiszen) [reply lib/zendesk_apps_support/package.rb]: Yeah good call 👍 I'll go with `product_directory`

File: lib/zendesk_apps_support/assets/src.js.erb [hunk 2]
```diff
@@ -21,6 +21,7 @@ var app = ZendeskApps.defineApp(<%= iframe_only ? 'null' : 'source' %>)
     appVersion: <%= version.to_json %>,
     locationIcons: <%= location_icons.to_json %>,
     assetUrlPrefix: <%= asset_url_prefix.to_json %>,
+    logoAssetHash: <%= logo_asset_hash.to_json %>,
```
Comment (danielbreves) [inline lib/zendesk_apps_support/assets/src.js.erb]: @mtsolakiszen do we have a plan to regenerate these assets? src.js is normally only updated when the app is updated.

Comment (mtsolakiszen) [reply lib/zendesk_apps_support/assets/src.js.erb]: @danielbreves The implementation has a fallback to use the current logic for retrieving the logo if the `logo_assets_hash` isn't defined therefore new apps will use the new method and all existing apps should still work

**Comment by danielbreves** [comment]
Will this simply 404 on ZAT if a multi-product app doesn't have a product folder?

**Comment by seangoedecke** [comment]
Dev QAing this PR, based on @waferbaby's work:

For a single product app, installed.js contains the `logoAssetHash` with appropriate contents:
```
function() {
    var app = ZendeskApps.defineApp(null)
  .reopenClass({"experiments":{},"location":{"support":{"ticket_sidebar":{"url":"assets/iframe.html"}}},"noTemplate":[],"singleInstall":false,"signedUrls":false})
  .reopen({
    appName: "Buddha Machine",
    appVersion: "1.0",
    locationIcons: {},
    assetUrlPrefix: "/api/v2/apps/61/assets/",
    logoAssetHash: {"support":"logo-small.png"},
    appClassName: "app-61",
    author: {
      name: "Zendecks",
      email: "support@zendecks.com"
    },
    frameworkVersion: "2.0"
  });

ZendeskApps["Buddha Machine"] = app;


    if (ZendeskApps["Buddha Machine"]) {
      ZendeskApps["Buddha Machine"].install({"id":10001,"app_id":61,"app_name":"Buddha Machine","enabled":true,"settings":{"name":"Buddha Machine","title":"Buddha Machine"},"requirements":[],"updated_at":"2018-01-08T04:09:20Z","created_at":"2018-01-08T04:09:20Z"});
    }

    ZendeskApps.sortAppsForSite("ticket_sidebar", [10001]);
}());


ZendeskApps.trigger && ZendeskApps.trigger('ready');
```

For a multi product app, `logoAssetHash` contains logos for multiple products:
```
var app = ZendeskApps.defineApp(null)
  .reopenClass({"experiments":{},"location":{"support":{"ticket_sidebar":{"url":"assets/iframe.html"}},"chat":{"chat_sidebar":{"url":"assets/iframe.html"}}},"noTemplate":[],"singleInstall":false,"signedUrls":false})
  .reopen({
    appName: "Nada Buddha Machine",
    appVersion: "1.0",
    locationIcons: {},
    assetUrlPrefix: "/api/v2/apps/62/assets/",
    logoAssetHash: {"support":"support/logo-small.png","chat":"chat/logo-small.png"},
    appClassName: "app-62",
    author: {
      name: "cgoddard",
      email: "cgoddard@zendesk.com"
    },
    frameworkVersion: "2.0"
  });
```

**Pull request merged**